src/Storages/Streaming/storageUtil.cpp

```diff
@@ -26,4 +29,23 @@ String getStorageName(const ASTCreateQuery & create)
     else
         return "Stream";
 }
+
+bool isStreamingStorage(const StoragePtr & storage, ContextPtr context)
```

Comment:
Why do not modify the logic of ProxyStream::isStreaming()? Why an extra function is added here?

Reply:
There are different storages (ProxyStream is only one of them), such as `Stream', 'Random stream', 'MaterializedView' .etc , in addition, it depends on setting `query_mode` in Context.
So we need a common function to check